### ISSUE TITLE
Add adahessian

### ISSUE THREAD

**Issue by bratao** [jettify/pytorch-optimizer#169]
Hello, 

I´m a big fan of this project. Recently a new optimized has been proposed, that promises SOTA for many tasks.

https://github.com/amirgholami/adahessian

The possibility of using it here is very good!

**Comment by jettify** [comment]
Thanks! I will take a look.

**Comment by jettify** [comment]
Since authors licensed code under unfriendly GPL V3, I can not reuse that code here and unfortunitly have to reproduce new implementation from paper.

Commercial usage of of authors implementation is limited, since forces users to license potentially commercial code under GPL also.

**Comment by bratao** [comment]
@jettify this is a re-implementation from scratch of the paper, with a MIT license
https://github.com/davda54/ada-hessian

**Comment by amirgholami** [comment]
Hi @jettify,

First, thanks so much for creating this repository, it is a great resource. I am one of the Adahessian authors, and we have just changed the license to the more friendly MIT License. I also just opened a new pull request that adds Adahessian optimizer to your repository:

https://github.com/jettify/pytorch-optimizer/pull/239#issue-538702447

I got the following results when I ran the optimizer on the  Rosenbrock and Rastrigin test:



- `Rosenbrock: best loss: 0.002011366948950946,  best lr=  32.26606775899826`
- `Rastrigin: best loss: 2.3379755056396334e-05, best lr=  23.15826699019325



I have also attached the visualization for each test:

![rosenbrock_Adahessian](https://user-images.githubusercontent.com/10134507/102001866-8138f580-3cab-11eb-83b2-1c81a500e4e8.png)
![rastrigin_Adahessian](https://user-images.githubusercontent.com/10134507/102001868-84cc7c80-3cab-11eb-80ff-aca2b3205268.png)

P.S: Based on the results I got, it seems Adahessian gets the best loss value among the optimizers

P.S2: @bratao Thanks so much for adding the Adahessian optimizer and for promoting it.
